core/src/test/java/org/elasticsearch/search/suggest/SuggestBuilderTests.java

```diff
@@ -87,7 +88,7 @@ public void testFromXContent() throws IOException {
      */
     public void testEqualsAndHashcode() throws IOException {
         for (int runs = 0; runs < NUMBER_OF_RUNS; runs++) {
-            checkEqualsAndHashCode(randomSuggestBuilder(), original -> {
+            EqualsHashCodeTestUtils.<SuggestBuilder>checkEqualsAndHashCode(randomSuggestBuilder(), original -> {
```

Comment:
When making these changes please add a comment that it is here to workaround compiler bugs, and link to an existing compiler bug or open a new one. Without a comment, someone else might see the generic parameter that is not needed per the specification and turn around and remove it and we are back to Eclipse being broken. Without an upstream issue to track and linked to from the comment we do not know when it's safe to remove.